src/traces/bar/defaults.js

```diff
@@ -77,5 +76,54 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
     errorBarsSupplyDefaults(traceIn, traceOut, lineColor || Color.defaultLine, {axis: 'y'});
     errorBarsSupplyDefaults(traceIn, traceOut, lineColor || Color.defaultLine, {axis: 'x', inherit: 'y'});
 
+    handleGroupingDefaults(traceIn, traceOut, layout, coerce);
+
     Lib.coerceSelectionMarkerOpacity(traceOut, coerce);
+}
+
+function handleGroupingDefaults(traceIn, traceOut, layout, coerce) {
+    var orientation = traceOut.orientation;
+    // TODO make this work across matching axes too?!?
+    // TODO should this work per trace-type?
+    //      one set for bar/histogram another for box/violin?
+    //      or just one set for all trace trace types?
```

Comment:
Does anyone here have a strong opinion about this? 

Should `alignmentgroup` and `offsetgroup` work across trace types? Or should we match bar and histogram and box and violins separately?

Reply:
Seems like making it work across trace types would be the most flexible - you can always just give different values to different trace types, which I guess is what the default would be. It's also probably what I'd expect as a user. It's an awfully weird use case, so not worth a huge effort, but if we omit it now and wanted to add it later it would be a breaking change.

Reply:
Bars above for counts (say) and boxes below for distributions? Seems not that weird to me :) kinda useful actually!

Reply:
Ah yeah good point. That would magically work if you have the same traces in each case, and in the same order... but break those assumptions (which could happen for good reasons) and you're back to manual alignment if we don't implement this cross-type.

Reply:
Done in https://github.com/plotly/plotly.js/pull/3529/commits/e746b088d6ff929577416706f59c2d739f850640